src/modules/kvs/kvstxn.c

```diff
@@ -63,6 +63,14 @@ struct kvstxn {
     zlist_t *dirty_cache_entries_list;
     int internal_flags;
     kvstxn_mgr_t *ktm;
+    /* State transitions
+     *
+     * INIT -> LOAD_ROOT
+     * LOAD_ROOT -> APPLY_OPS
+     * APPLY_OPS -> STORE
+     * STORE -> PRE_FINISHED
+     * PRE_FINISHED -> FINISHED
+     */
```

Comment:
Since the state machine is _linear_ how about just:
```
INIT -> LOAD_ROOT -> APPLY_OPS -> STORE -> PRE_FINISHED -> FINISHED
```
Would it be useful to have a sentence or two describing each state also?

Reply:
yeah, i can write a bit about each state.  I'll keep the separate on each line though, b/c it was with knowledge of non-linear in a future commit.